api/cloud-resources/v1beta1/gcpredisinstance_types.go

```diff
@@ -59,23 +85,16 @@ type MaintenancePolicy struct {
 }
 
 // GcpRedisInstanceSpec defines the desired state of GcpRedisInstance
-// +kubebuilder:validation:XValidation:rule=(self.tier == "BASIC" && self.replicaCount == 0 || self.tier == "STANDARD_HA"), message="replicaCount must be zero for BASIC tier"
-// +kubebuilder:validation:XValidation:rule=(self.tier == "STANDARD_HA" && self.replicaCount > 0 || self.tier == "BASIC"), message="replicaCount must be defined with value between 1 and 5 for STANDARD_HA tier"
-// +kubebuilder:validation:XValidation:rule=(self.tier == "STANDARD_HA" && self.memorySizeGb >= 5 || self.tier == "BASIC"), message="memorySizeGb must be at least 5 GiB for STANDARD_HA tier"
+// +kubebuilder:validation:XValidation:rule=(self.redisTier.startsWith('S') && self.replicaCount == 0 || self.redisTier.startsWith('P')), message="replicaCount must be zero for Standard redisTier offering"
+// +kubebuilder:validation:XValidation:rule=(self.redisTier.startsWith('P') && self.replicaCount > 0 || self.redisTier.startsWith('S')), message="replicaCount must be defined with value between 1 and 5 for Premium redisTier offering"
 type GcpRedisInstanceSpec struct {
 
 	// +optional
 	IpRange IpRangeRef `json:"ipRange"`
 
-	// The service tier of the instance.
-	// +kubebuilder:default=BASIC
-	// +kubebuilder:validation:XValidation:rule=(self == oldSelf), message="Tier is immutable."
-	// +kubebuilder:validation:Enum=BASIC;STANDARD_HA
-	Tier string `json:"tier"`
-
-	// Redis memory size in GiB.
 	// +kubebuilder:validation:Required
-	MemorySizeGb int32 `json:"memorySizeGb"`
+	// +kubebuilder:validation:XValidation:rule=(self.startsWith('S') && oldSelf.startsWith('S') || self.startsWith('P') && oldSelf.startsWith('P')), message="redisTier category cannot be changed."
+	RedisTier GcpRedisTier `json:"redisTier"`
```

Comment:
This will allow going from S1 to S2 and similar. Guess that's what you wanted. If so the message is not quite appropriate since S1 to S2 is a change.

Reply:
Thats what I wanted.
The "category" in the message refers to standard and premium.
You should be able to move within capacity tiers inside a category, but you should not be able to switch a category itself.
The examples are in api-tests.

Do you have an idea how to write the message to be more descriptive to the user?

Reply:
Going to change "category" to "service tier" as per team meeting